### PR TITLE
Sort HW builder's chapter sections so they appear in order

### PR DESCRIPTION
**Pull request by nathanstitt**
screen shot 2015-07-28 at 3 13 03 pm

### CODE REVIEW

**Comment by nathanstitt** [comment]
I somehow used the wrong branch name on this, hence the name "progress-bar-color".  I didn't notice until I'd opened the PR, so "A rose by any other name would smell as sweet" (or stinky as it might be)

File: src/flux/task-plan.coffee [hunk 1]
```diff
@@ -20,7 +20,7 @@ PLAN_TYPES =
 sortTopics = (topics) ->
   _.sortBy(topics, (topicId) ->
     topic = TocStore.getSectionInfo(topicId)
-    topic.chapter_section.toString()
+    TaskHelpers.chapterSectionToNumber(topic.chapter_section)
```
Comment (philschatz) [inline src/flux/task-plan.coffee]: I do sometimes miss `java.lang.Comparator`

File: test/task-helpers.spec.coffee [hunk 2]
```diff
@@ -25,3 +25,8 @@ describe 'TaskHelpers', ->
     )
     expect( lateness.is_late ).to.be.true
     expect( lateness.how_late ).to.equal('30 minutes')
+
+  it 'converts chapter_sections to numbers', ->
+    expect(Helpers.chapterSectionToNumber([1])).to.equal(100)
+    expect(Helpers.chapterSectionToNumber([1, 2])).to.equal(102)
+    expect(Helpers.chapterSectionToNumber([31, 88, 42])).to.equal(318842)
```
Comment (philschatz) [inline test/task-helpers.spec.coffee]: Do we need to worry/test `0`'s?

```coffee
expect(Helpers.chapterSectionToNumber([1, 0, 2])).to.equal(10002)
expect(Helpers.chapterSectionToNumber([1, 0])).to.equal(100) # not sure
expect(Helpers.chapterSectionToNumber([1])).to.equal(1) # not sure
```

Comment (nathanstitt) [inline test/task-helpers.spec.coffee]: Probably not.  The algorithm doesn't do anything special to handle them but the test doesn't need to explicitly test them

**Pull request merged**